lib/entry/crossDCsStreamPersistence.js

```diff
@@ -0,0 +1,191 @@
+            return true;
+        } else {
+            // keep the playlistGenerator obj
+            this._entriesData[entryId] = { playlist: playlistGenerator, timestamp: new Date() };
+            this._entriesToProcess[entryId] = entryId;
+            return this.flushEntryList(this._entriesToProcess);
+        }
+
+    });
+};
+
+CrossDCsStreamPersistence.prototype.removeEntry = function (entryId) {
+    var that = this;
+    return Q.fcall(function () {
+        var timeoutObj = that._entriesToProcess[entryId];
+        if (timeoutObj) {
+            logger.debug(`aborting entry [${entryId}] persist timer`);
+            clearTimeout(timeoutObj);
+            delete that._entriesToProcess[entryId];
+            delete that._entriesData[entryId];
+            return that.flushEntryList(that._entriesToProcess);
+        }
+    });
+};
+
+// initialized persistence list.
+CrossDCsStreamPersistence.prototype.init = function () {
+
+    let deferred = Q.defer();
+    let entries;
+
+   qio.exists(stoppedEntriesPath)
+        .then((exists) => {
+            if (exists) {
+                return qio.read(this.stoppedEntriesFileName);
+            } else {
+                qio.makeDirectory(stoppedEntriesPath)
+                    .then(() => {
+                        return this.backwardCompatibleInit();
+                    });
+            }
+        })
+        .then((stoppedEntries) => {
+               if (stoppedEntries) {
+                   _.each(JSON.parse(stoppedEntries), (entryId) => {
+                       this.addEntry(new PlaylistGenerator({entryId: entryId}, false));
+                       this._entriesData[entryId].playlist.restoreStopped();
+                   });
+               }
+            deferred.resolve();
+        })
+        .catch((err) => {
+            logger.error(`error restoring stopped entries list, error: [${ErrorUtils.error2string(err)}]`);
+            deferred.reject(err);
+        });
+
+    return deferred.promise;
+};
+
+CrossDCsStreamPersistence.prototype.flushEntryList = function (stoppedEntriesList) {
+
+    let stoppedEntriesStr = JSON.stringify(_.keys(stoppedEntriesList));
+
+    return qio.write(this.stoppedEntriesFileName, stoppedEntriesStr)
+        .then(() => {
+          logger.debug(`Stopped entries: ${stoppedEntriesStr}`);
+        })
+        .catch((err) => {
+            logger.error(`failed to save stopped entries persistence data, error: [${ErrorUtils.error2string(err)}]`);
+        });
+};
+
+CrossDCsStreamPersistence.prototype.processStoppedEntries = function() {
+    let now = Date.now;
+    _.each(this._entriesToProcess, (entryId) => {
+        if (now - this._entriesData[entryId].timestamp >= this.checkStreamIntervalMSec) {
+            this.streamPersistenceCheck(entryId);
+        }
+    });
+
+    return this.flushEntryList(this._entriesToProcess);
+
+};
+
+CrossDCsStreamPersistence.prototype.backwardCompatibleInit = function() {
```

Comment:
why do we need this? we don't care about the transition ... it doesn't worth even 5 seconds...